src/gui/editors/AutomationEditor.cpp

```diff
@@ -663,63 +637,89 @@ void AutomationEditor::mouseReleaseEvent(QMouseEvent * mouseEvent )
 		m_mouseDownLeft = false;
 		mustRepaint = true;
 	}
-	if ( mouseEvent->button() == Qt::RightButton )
+	if (mouseEvent->button() == Qt::RightButton)
 	{
 		m_mouseDownRight = false;
 		mustRepaint = true;
 	}
 
-	if( mouseEvent->button() == Qt::LeftButton )
-	{
-		mustRepaint = true;
-	}
-
-	if( m_editMode == DRAW )
+	if (m_editMode == DRAW)
 	{
-		if( m_action == MOVE_VALUE )
+		if (m_action == MOVE_VALUE)
 		{
+			// Actually apply the value of the node being dragged
 			m_pattern->applyDragValue();
 		}
+
 		QApplication::restoreOverrideCursor();
 	}
 
 	m_action = NONE;
 
-	if( mustRepaint )
-	{
-		repaint();
-	}
+	if (mustRepaint) { repaint(); }
 }
 
 
 
 
-void AutomationEditor::removePoints( int x0, int x1 )
+/**
+ * @brief Removes all automation nodes between the given ticks
+ * @param Int first tick of the range
+ * @param Int second tick of the range
+ */
+void AutomationEditor::removeNodes(int tick0, int tick1)
 {
-	int deltax = qAbs( x1 - x0 );
-	int x = x0;
-	int xstep;
+	if (tick0 == tick1) { return; }
 
-	if( deltax < 1 )
-	{
-		return;
-	}
+	TimePos start = TimePos(qMin(tick0, tick1));
+	TimePos end = TimePos(qMax(tick0, tick1));
 
-	if( x0 < x1 )
+	timeMap & tm = m_pattern->getTimeMap();
+	timeMap::iterator it = tm.lowerBound(start);
+
+	// Make a list of TimePos with nodes to be removed
+	// because we can't simply remove the nodes from
+	// the timeMap while we are iterating it.
+	QVector<TimePos> nodesToRemove;
+
+	while (it != tm.end())
 	{
-		xstep = 1;
+		if (POS(it) > end) { break; }
+
+		nodesToRemove.append(POS(it));
+		++it;
 	}
```

Comment:
You can use a for-loop like this:
```cpp
for (auto it = tm.lowerBound(start), endIt = tm.upperBound(end); it != endIt; ++it)
{
	nodesToRemove.append(POS(it));
}
```
Here `tm.upperBound(end)` is used because the range is `[start, end]`. If it was `[start, end)`, `tm.lowerBound(end)` would be used instead.

Reply:
You can also apply this to other places where you work on specific ranges.

Reply:
Done! I'll now work on moving those methods to `AutomationPattern`!